server/service/service.go

```diff
@@ -404,7 +405,64 @@ func (s *Service) BootstrapStream(stream bpb.Bootstrap_BootstrapStreamServer) er
 			}
 
 		case *bpb.BootstrapStreamRequest_Response_:
-			// TODO: process response
+			log.Infof("Received Response from %s", deviceID)
+			if session.currentState != stateTPM20NonceSent {
+				return status.Errorf(codes.FailedPrecondition, "unexpected state %v for device %s, expecting nonce response", session.currentState, deviceID)
```

Comment:
For this, I would use invalid argument. Failed precondition implies the server is in the wrong state but in this context it's because the device is in the wrong state.